mods/beds/functions.lua

```diff
@@ -81,9 +87,27 @@ local function lay_down(player, pos, bed_pos, state, skip)
 
 	-- lay down
 	else
+
 		beds.pos[name] = pos
+		meta = minetest.get_meta(beds.pos[name])
+
+		-- Check if bed is occupied
+		for _, other_pos in pairs(beds.bed_position) do
+			if meta:get_string("occupiedby") ~= nil then
```

Comment:
The for loop around this is redundant

Reply:
Oh it is? I'll remove it.

Reply:
`meta` does not change, so you are checking the same thing every iteration
the loop veriable `other_pos` is also entirely unused

Reply:
Doesn't work without the loop; it says the bed is occupied even though it isn't.